osu.Game/Skinning/SkinManager.cs

```diff
@@ -307,7 +307,7 @@ public void EnsureMutableSkin()
 
         public void Save(Skin skin)
         {
-            if (skin.SkinInfo.ID <= 0)
+            if (!skin.SkinInfo.IsManaged)
```

Comment:
I'm re-reading this multiple times again and I am having an unshakable feeling that `IsValid` would be the more correct property to use in contexts such as this. In the context of realm specifically, wherever the question is "does this model correspond to a database row", it seems like it would be safer to use the value that says "this is a DB model, it wasn't deleted, and it won't explode on you if you touch it" (`IsValid`), rather than "this was a DB model at some point but it is no longer really certain if it still is" (`IsManaged`). Although then again, maybe letting a deleted model even reach this point would be considered programmer error?

I dunno, I'm having a difficult time getting my head round this stuff for some reason. It's not a huge point, I'm fine with merging as-is.

Reply:
Ah, I thought I posted a comment here. General gist was that I agree `IsValid` is better for correctness, but `IsManaged` is _acceptable_ so I'm with rolling with it for now.